content/blog/2025-05-27-introducing-enhanced-json-capabilities-in-valkey.md

```diff
@@ -0,0 +1,191 @@
+
+That changes with the general availability of native JSON support in Valkey. You can now store, query, and update JSON documents directly, without manual parsing or transformation. This brings a cleaner model to working with semi-structured data and makes your code easier to write and maintain.
+
+In this post, we'll guide you through setting up the Valkey JSON module and demonstrate how to use it for some common workloads.
+
+## Core Capabilities and Performance
+
+Valkey’s JSON support goes beyond basic storage and retrieval. It introduces powerful querying and filtering capabilities that streamline complex data operations. Tasks that once required extensive development time and multiple network requests can now be executed through a single JSON command, significantly increasing developer productivity and reducing operational costs.
+
+Valkey JSON supports six data types—**null, boolean, number, string, object, and array**—allowing developers to represent complex, nested data structures without the constraints of traditional string serialization. Unlike other composite types, JSON objects and arrays in Valkey can contain any combination of the six value types, enabling deeply nested and recursive data models to be stored natively.
+
+Internally, Valkey JSON utilizes an optimized **binary tree-like format**, which enables rapid traversal and manipulation of substructures without requiring full document parsing. This structure not only minimizes memory usage but also ensures that operations on specific paths remain efficient. Path-based commands like `JSON.GET`, `JSON.SET`, and `JSON.DEL` allow targeted interactions with specific elements, supporting multiple paths within a single operation. Additionally, Valkey JSON integrates with Valkey’s Access Control Lists (ACLs), introducing a `@json` command category to enforce granular permissions alongside existing data types.
+
+## Getting Started with Valkey JSON
+
+Let’s walk through a practical example using Valkey JSON with the [`valkey-bundle`](https://hub.docker.com/r/valkey/valkey-bundle) Docker image.
+
+### Installation and Setup
+
+Valkey JSON comes pre-loaded in the `valkey-bundle`, which also includes other modules such as Valkey Bloomfilters, Valkey search and Valkey LDAP loaded on Valkey.
+
+#### Start a Valkey bundle instance
+```
+docker run --name my-valkey-bundle -d valkey/valkey-bundle
+```
+This will start a docker container of Valkey with JSON module already loaded.
+
+#### Connect with valkey-cli
+To connect to your running instance using the built-in valkey-cli:
+```
+docker run -it --network container:my-valkey-bundle --rm valkey/valkey-bundle valkey-cli
+127.0.0.1:6379>
+```
+Tip: --network container:my-valkey-bundle allows the CLI to communicate directly with the same container.
```

Comment:
I still think you can just remove the network argument and it should still work.